File: csharp/56-merge-intervals.cs
public class Solution {
    public int[][] Merge(int[][] intervals) {
        var sortedIntervals = intervals.Clone() as int[][];
        Array.Sort(sortedIntervals, (a, b) => a[0] - b[0]);

        var mergedIntervals = new List<int[]>();
        var lastInterval = sortedIntervals[0];
        mergedIntervals.Add(lastInterval);

        for (var i = 1; i < sortedIntervals.Length; i++) {
            if (lastInterval[1] >= sortedIntervals[i][0]) {
                lastInterval[1] = Math.Max(lastInterval[1], sortedIntervals[i][1]);
            } else {
                lastInterval = sortedIntervals[i];
                mergedIntervals.Add(lastInterval);
            }
        }
        return mergedIntervals.ToArray();
    }
}

File: rust/56-merge-intervals.rs
impl Solution {
    pub fn merge(mut intervals: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        intervals.sort_by(|a, b| a[0].cmp(&b[0]));
        let res = vec![intervals.first().unwrap().clone()];
        intervals.into_iter().skip(1).fold(res, |mut res, curr| {
            let last = res.last().unwrap();
            if curr[0] <= last[1] {
                res.last_mut().unwrap()[1] = last[1].max(curr[1]);
            } else {
                res.push(curr)
            }
            res
        })
    }
}